### DISCUSSION TITLE
Additional useful methods?

### DISCUSSION THREAD

**Discussion by Agent6-6-6** [robbievanleeuwen/concrete-properties#57] [Ideas]
@robbievanleeuwen in working through a moment-curvature example for the NZS3101 code I've come across a couple of things I think should/could be added as general methods:-

1. A method to report the moment capacity (or curvature) at which the extreme tension bar first hits yield strain (or any other strain) using serviceability or ultimate strain profile.
    - Reason for adding is this particular point is required from a perspective of determining the effective stiffness to use in a seismic analysis when tension bars hit first yield (effective stiffness can be determined from the cracked second moment of area based on reaching this point of first yield)
    - Or in a moment-curvature analysis to create a simplified bilinear approximation for the true moment-curvature relationship by joining a line through the first yield moment/curvature point, and intersecting this with a horizontal line through the max moment capacity:-
    - something like this
![image](https://user-images.githubusercontent.com/36808164/196365574-1ca2f318-b958-43a2-91e4-463bcfd2e6c4.png)
![image](https://user-images.githubusercontent.com/36808164/196366293-b6a8e9d8-4da2-4dc4-a5cf-0c0d78c07fe2.png)

2. A method to return the reinforcement 'effective depth' based on the centroid of the tension reinforcement force at any moment using any strain profile. Note it is usually defined in codes as a centroid of tension reinforcement area, but the proper implementation should really be the centroid of the tension force as some bars closer to the neutral axis may not be at the yield stress.
   - Useful for determining shear capacity which often uses the effective depth 'd' within design equations
   - Also useful for determining curvature capacity via code simplifications/empirical relationships, for example:-
   - 
![image](https://user-images.githubusercontent.com/36808164/196366817-3da54836-8ed6-4d3f-b6da-71477388c905.png)

Looking through the various methods currently provided, there are bits and pieces of the puzzle but not having written these I'm not entirely sure where to start by stringing some existing bits together and/or writing similar. Any thoughts on implementing this @robbievanleeuwen as you're obviously the most familiar with all the background workings to know the potential best way!?

**Comment by connorferster** [comment]
@Agent6-6-6 

I think these would be really useful methods, too

**Reply by Agent6-6-6** [reply]
Another idea for a future enhancement ... 

When reviewing concrete structures for deflection, especially if doing ballpark prelim sizing, I've found it useful in the past to generate a curve based on moment vs cracked/effective moment of inertia for a given section and reinforcement layout. That way from the analysis moment you can read off or interpolate roughly the actual cracked stiffness, and update the analysis stiffness. Also useful for feeding the curve to Geotech if they are undertaking a wall design for you for example on concrete piles. Get them to do the iterative work for you!
